### PR TITLE
Fix; make expect(1).toBeGreaterThan(2) throw reliably

### PR DESCRIPTION
**Pull request by argosphil**
### What does this PR do?

This changes expect(...).toBeGreaterThan() to throw an exception when appropriate even when ANSI colors are not in use.

I noticed this while debugging #8833 / #8834 , and it prevented me from writing a test for it so far.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

I ran a local test, which falsely succeeded before the change and failed after it.

- [x] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [ ] I included a test for the new code, or an existing test covers it
- [x] JSValue used outside outside of the stack is either wrapped in a JSC.Strong or is JSValueProtect'ed
- [ ] I wrote TypeScript/JavaScript tests and they pass locally (`bun-debug test test-file-name.test`)

### CODE REVIEW

File: src/bun.js/test/expect.zig [hunk 1]
```diff
@@ -1774,6 +1775,7 @@ pub const Expect = struct {
             globalObject.throw(comptime Output.prettyFmt(fmt, true), .{ expected_fmt, value_fmt });
             return .zero;
         }
+	globalObject.throw(comptime Output.prettyFmt(fmt, false), .{ expected_fmt, value_fmt });
```
Comment (Electroid) [inline src/bun.js/test/expect.zig]: Please make sure to run the Zig formatter. If you use VSCode and our recommended extension it should do it for you.

Also, could we add some tests in `test/js/bun/test` to make sure this doesn't regress?

Comment (Electroid) [reply src/bun.js/test/expect.zig]: Actually, I think we can pass on this test for now. it's a bit silly to test for this specifically. later we should add overall tests to check output with ansi, ansi + color, color, and none; but that's out of scope for this pr.

Comment (argosphil) [reply src/bun.js/test/expect.zig]: Thanks for the zig formatter reminder, and sorry for not knowing about it.

I agree about the tests, FWIW. Adding a NO_COLOR option to the test matrix would also work.

**Review by Electroid — APPROVED** [review]
Thanks!

**Pull request merged**